sdk/storage/azure-storage-blobs/src/blob_batch_client.cpp

```diff
@@ -105,7 +105,10 @@ namespace Azure { namespace Storage { namespace Blobs {
     }
     policies.emplace_back(std::make_unique<StoragePerRetryPolicy>());
     policies.emplace_back(std::make_unique<SharedKeyPolicy>(credential));
-    policies.emplace_back(std::make_unique<Azure::Core::Http::TransportPolicy>());
+    Azure::Core::Http::TransportPolicyOptions transportPolicyOptions;
```

Comment:
Make this option the actual `option` inside the ClientOptions instead of creating it here and passing the value.
The current pattern here is `mapping an client option to an azure core option` which we never want to have.
Client is expected to expose Azure Core options structures directly, just like the retry options or any  other policy option structure.